### ISSUE TITLE
VoiceOver inside intellisense is a bit of a mess

### ISSUE THREAD

**Issue by Tyriar** [Microsoft/vscode#40597]
Some problems I've experienced while using voiceover:

<img width="1407" alt="screen shot 2017-12-20 at 11 31 09 am" src="https://user-images.githubusercontent.com/2193314/34224948-7935cc32-e579-11e7-8e9c-301b88bda12f.png">
<img width="1171" alt="screen shot 2017-12-20 at 11 31 34 am" src="https://user-images.githubusercontent.com/2193314/34224951-7be97ee2-e579-11e7-9c4f-069ddd72f690.png">
<img width="1190" alt="screen shot 2017-12-20 at 11 34 20 am" src="https://user-images.githubusercontent.com/2193314/34225013-be1c3264-e579-11e7-8005-009691f733cc.png">

- "Object object" is being read out after every intellisense item
- The intellisense detail is being read out immediately after the list item, which means a short description is read followed by a long description (which may be the same)
  - "private method private method definition"
  - "public public"
  - "true true"
- Snippets are not announced as snippets?

@mjbvz related: is "typescript support for vscode needed"? The user knows they're in VS Code.

**Comment by mjbvz** [comment]
The TS extension does not add the `(typescript support for vscode)` section. I think that comes from `SnippetSuggestion`: https://github.com/Microsoft/vscode/blob/master/src/vs/workbench/parts/snippets/electron-browser/snippetsService.ts#L288
